### PR TITLE
Add documentation on enabling Microsoft.Insights for Azure Events / Metrics

### PR DESCRIPTION
**Pull request by agrare**
The Microsoft.Insights resource provider is required for ManageIQ Events and Metrics features to be able to work with an Azure cloud provider.

Without this resource provider enabled events / metrics will be disabled.

image

This has always been a requirement for azure events/metrics, we had a talk article from a community member about how to enable this resource provider but this doesn't appear to be viewable any longer https://talk.manageiq.org/t/how-can-i-see-azure-utilization/5298/4
It doesn't seem like instructions on this being a requirement, or how to enable it ever made it to the documentation.

### CODE REVIEW

**Comment by agrare** [comment]
https://github.com/ManageIQ/manageiq-providers-azure/issues/561 got me looking in the documentation for where we instruct users on how to enable this, and sure enough it was never added to the docs.

**Review by Fryguy — APPROVED** [review]

**Comment by Fryguy** [comment]
@qahmed1998 please review

**Comment by agrare** [comment]
Completely forgot this was still open 😆

**Comment by Fryguy** [comment]
Backported to `radjabov` in commit 8c27432bed4a29e69d02ae1fd0f24f34d26048e2.

```text
commit 8c27432bed4a29e69d02ae1fd0f24f34d26048e2
Author: Jason Frey <fryguy9@gmail.com>
Date:   Thu Dec 19 09:25:42 2024 -0500

    Merge pull request #1811 from agrare/add_azure_microsoft_insights_provider_instructions
    
    Add documentation on enabling Microsoft.Insights for Azure Events / Metrics
    
    (cherry picked from commit 11241f9e666c9aa3cf0d7ad3bccd9939b44ed7d4)
```

**Comment by qahmed1998** [comment]
Sorry only saw this now. And didn't get a notification. The doc that is added needs to be edited.

File: managing_providers/cloud_providers/_topics/adding_azure_providers.md [hunk 1]
```diff
@@ -62,6 +62,15 @@ So, after a service principal account (instance of an application in a directory
 
 You can now use these values in the procedure below to add an Azure cloud instance as a provider to {{ site.data.product.title_short }}.
 
+In order to be able to capture Events or Capacity & Utilization metrics from your Azure account
+you first have to register the Microsoft Insights resource provider.
+
+1. Select your subscription, and click on "Resource providers"
```
Comment (qahmed1998) [inline managing_providers/cloud_providers/_topics/adding_azure_providers.md:68]: and click **Resource providers**.

File: managing_providers/cloud_providers/_topics/adding_azure_providers.md [hunk 2]
```diff
@@ -62,6 +62,15 @@ So, after a service principal account (instance of an application in a directory
...
+2. Enter "microsoft.insights" in the search box
```
Comment (qahmed1998) [inline managing_providers/cloud_providers/_topics/adding_azure_providers.md:69]: **microsoft.insights** in the search box.

File: managing_providers/cloud_providers/_topics/adding_azure_providers.md [hunk 3]
```diff
@@ -62,6 +62,15 @@ So, after a service principal account (instance of an application in a directory
...
+3. If the Provider isn't registered, then hit the checkbox and select the Register option
```
Comment (qahmed1998) [inline managing_providers/cloud_providers/_topics/adding_azure_providers.md:70]: If the provider isn't registered, then click on checkbox and select the **Register** option.

File: managing_providers/cloud_providers/_topics/adding_azure_providers.md [hunk 4]
```diff
@@ -62,6 +62,15 @@ So, after a service principal account (instance of an application in a directory
...
+
+[Microsoft Insights Resource Provider](../../images/azure-microsoft-insights.png)
+
 **To Add an Azure Cloud Provider**:
```
Comment (qahmed1998) [inline managing_providers/cloud_providers/_topics/adding_azure_providers.md:74]: **Add an ....

File: managing_providers/cloud_providers/_topics/adding_azure_providers.md [hunk 5]
```diff
@@ -62,6 +62,15 @@ So, after a service principal account (instance of an application in a directory
...
 
 1.  Browse to menu: **Compute > Clouds > Providers**.
```
Comment (qahmed1998) [inline managing_providers/cloud_providers/_topics/adding_azure_providers.md:76]: Browse to the menu and click **Compute** > **Clouds** > **Providers**.

File: managing_providers/cloud_providers/_topics/adding_azure_providers.md [hunk 6]
```diff
@@ -62,6 +62,15 @@ So, after a service principal account (instance of an application in a directory
 
 You can now use these values in the procedure below to add an Azure cloud instance as a provider to {{ site.data.product.title_short }}.
 
+In order to be able to capture Events or Capacity & Utilization metrics from your Azure account
```
Comment (qahmed1998) [inline managing_providers/cloud_providers/_topics/adding_azure_providers.md:65]: To capture Events or Capacity and Utilization metrics...

**Pull request merged**